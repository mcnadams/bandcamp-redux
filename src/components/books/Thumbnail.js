import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Thumbnail({ isbn }) {
  return (
    <Link to={`/${isbn}`}>
      <img src={`http://covers.openlibrary.org/b/isbn/${isbn}-M.jpg`}/>
    </Link>
  );
}

Thumbnail.propTypes = {
  isbn: PropTypes.string.isRequired
};

export default Thumbnail;
