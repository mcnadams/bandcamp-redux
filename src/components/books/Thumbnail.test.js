import { shallow } from 'enzyme';
import React from 'react';
import Thumbnail from './Thumbnail';

describe('Thumbnail component', () => {
  it('renders thumbnail', () => {
    const wrapper = shallow(<Thumbnail isbn={'222'}/>);

    expect(wrapper).toMatchSnapshot();
  });
});
