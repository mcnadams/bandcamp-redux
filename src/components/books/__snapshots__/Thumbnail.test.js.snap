// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail component renders thumbnail 1`] = `
<Link
  to="/222"
>
  <p>
    222
  </p>
  <img
    alt="Image not found"
    src="http://covers.openlibrary.org/b/isbn/222-M.jpg"
  />
</Link>
`;
